programs/clearing_house/src/controller/bank_balance.rs

```diff
@@ -60,28 +80,112 @@ pub fn update_bank_cumulative_interest(bank: &mut Bank, now: i64) -> ClearingHou
     let InterestAccumulated {
         deposit_interest,
         borrow_interest,
-        utilization,
     } = calculate_accumulated_interest(bank, now)?;
 
     if deposit_interest > 0 && borrow_interest > 1 {
-        bank.cumulative_deposit_interest = bank
-            .cumulative_deposit_interest
-            .checked_add(deposit_interest)
+        // borrowers -> lenders IF fee here
+        let deposit_interest_for_stakers = deposit_interest
+            .checked_mul(bank.total_reserve_factor as u128)
+            .ok_or_else(math_error!())?
+            .checked_div(BANK_INTEREST_PRECISION)
             .ok_or_else(math_error!())?;
 
-        bank.cumulative_borrow_interest = bank
-            .cumulative_borrow_interest
-            .checked_add(borrow_interest)
+        let deposit_interest_for_lenders = deposit_interest
+            .checked_sub(deposit_interest_for_stakers)
             .ok_or_else(math_error!())?;
-        bank.last_interest_ts = cast_to_u64(now)?;
+
+        if deposit_interest_for_lenders > 0 {
+            bank.cumulative_deposit_interest = bank
+                .cumulative_deposit_interest
+                .checked_add(deposit_interest_for_lenders)
+                .ok_or_else(math_error!())?;
+
+            bank.cumulative_borrow_interest = bank
+                .cumulative_borrow_interest
+                .checked_add(borrow_interest)
+                .ok_or_else(math_error!())?;
+            bank.last_interest_ts = cast_to_u64(now)?;
+
+            // add deposit_interest_for_stakers as balance for insurance_fund_pool
+            let token_amount = get_interest_token_amount(
+                bank.deposit_balance,
+                bank,
+                deposit_interest_for_stakers,
+            )?;
+
+            update_insurance_fund_pool_balances(token_amount, &BankBalanceType::Deposit, bank)?;
+        }
     }
 
-    update_bank_twap_stats(bank, utilization, now)?;
+    update_bank_twap_stats(bank, now)?;
     bank.last_twap_ts = cast_to_u64(now)?;
 
     Ok(())
 }
 
+pub fn update_insurance_fund_pool_balances(
```

Comment:
whyd you copy all this code?

Reply:
ah...

since the bank mut, could pass in the bank.insurance_fund_pool as a mut either, so did a quick lazy copy